### PR TITLE
Fix inequivalence of operators

### PR DESCRIPTION
**Pull request by martong**

### CODE REVIEW

**Comment by balazske** [comment]
Was this really not checked before?

File: lib/AST/ASTStructuralEquivalence.cpp [hunk 1]
```diff
@@ -1648,6 +1648,13 @@ bool StructuralEquivalenceContext::CheckKindSpecificEquivalence(
     }
   } else if (FunctionDecl *FD1 = dyn_cast<FunctionDecl>(D1)) {
     if (FunctionDecl *FD2 = dyn_cast<FunctionDecl>(D2)) {
+      if (FD1->isOverloadedOperator()) {
+        if (!FD2->isOverloadedOperator())
+          return false;
+        if (getOperatorSpelling(FD1->getOverloadedOperator()) !=
+            getOperatorSpelling(FD2->getOverloadedOperator()))
```
Comment (balazske) [inline lib/AST/ASTStructuralEquivalence.cpp]: Why not compare the value from `getOverloadedOperator` directly (instead of spelling)?

**Comment by balazske** [comment]
`IsStructurallyEquivalent`with `CXXMethodDecl` has a check for `getOverloadedOperator()` that can be removedif the new code is used.

File: lib/AST/ASTStructuralEquivalence.cpp (hunk 1 above)
Comment (martong) [reply lib/AST/ASTStructuralEquivalence.cpp]: Good point.

**Comment by martong** [comment]
> IsStructurallyEquivalentwith CXXMethodDecl has a check for getOverloadedOperator() that can be removedif the new code is used.

I don't think so, because when we have a `CXXMethodDecl` then we do not check the `FunctionDecl` related things:
```
  // ...
  } else if (auto *MD1 = dyn_cast<CXXMethodDecl>(D1)) {
    // ...
  } else if (FunctionDecl *FD1 = dyn_cast<FunctionDecl>(D1)) {
    // ...
  } else if (FriendDecl *FrD1 = dyn_cast<FriendDecl>(D1)) {
    // ...
  } // ...
```

**Comment by martong** [comment]
Update: Removed `getOperatorSpelling`.

**Review by balazske — APPROVED** [review]

**Comment by martong** [comment]
https://reviews.llvm.org/D57902

**Pull request merged**